plugin/audit_log/audit_log.c

```diff
@@ -775,6 +787,32 @@ int is_event_class_allowed_by_policy(unsigned int class,
 
 
 static
+void audit_log_update_thd_local(audit_log_thd_local *local,
+                                unsigned int event_class,
+                                const void *event)
+{
+  if (event_class == MYSQL_AUDIT_CONNECTION_CLASS)
+  {
+    const struct mysql_event_connection *event_connection=
+      (const struct mysql_event_connection *) event;
+    local->skip_logging= FALSE;
+    if (audit_log_include_accounts != NULL &&
+        !audit_log_check_account_included(event_connection->user,
+                                          event_connection->user_length,
+                                          event_connection->host,
+                                          event_connection->host_length))
+      local->skip_logging= TRUE;
```

Comment:
I added assert in `audit_log_include_accounts_update` and `audit_log_exclude_accounts_update`